agent/collector.go

```diff
@@ -169,79 +175,138 @@ func (l *Collector) postMessage(endpoint string, m model.MessageBody) {
 	if err != nil {
 		log.Errorf("Unable to encode message: %s", err)
 	}
-	l.cfg.APIEndpoint.Path = endpoint
-	url := l.cfg.APIEndpoint.String()
+
+	responses := make(chan postResponse)
+	for _, ep := range l.cfg.APIEndpoints {
+		go l.postToAPI(ep, checkPath, body, responses)
+	}
+
+	// Wait for all responses to come back before moving on.
+	statuses := make([]*model.CollectorStatus, 0, len(l.cfg.APIEndpoints))
+	for i := 0; i < len(l.cfg.APIEndpoints); i++ {
+		url := l.cfg.APIEndpoints[i].Endpoint.String()
+		res := <-responses
+		if res.err != nil {
+			log.Error(res.err)
+			continue
+		}
+
+		r := res.msg
+		switch r.Header.Type {
+		case model.TypeResCollector:
+			rm := r.Body.(*model.ResCollector)
+			if len(rm.Message) > 0 {
+				log.Errorf("error in response from %s: %s", url, rm.Message)
+			} else {
+				statuses = append(statuses, rm.Status)
+			}
+		default:
+			log.Errorf("unexpected response type from %s: %d", url, r.Header.Type)
+		}
+	}
+
+	if len(statuses) > 0 {
+		l.updateStatus(statuses)
+	}
+}
+
+func (l *Collector) updateStatus(statuses []*model.CollectorStatus) {
+	curEnabled := atomic.LoadInt64(&l.realTimeEnabled) == 1
+
+	// If any of the endpoints wants real-time we'll do that.
+	var nextEnable int
+	for _, s := range statuses {
```

Comment:
I'm a little confused by this. If we get two statuses back and the first is enabled, and the latter is  disabled, wont we set `nextEnable` to 0 causing RT mode to be disabled it?

Perhaps I'm missing something?

Reply:
If something is wrong, then maybe something like the following would work?

```
// If any of the endpoints wants real-time we'll do that.
shouldEnableRT := false
for _, s := range statuses {
  shouldEnableRT = shouldEnableRT || (s.ActiveClients > 0 && l.cfg.AllowRealTime)
}

if curEnabled && !shouldEnableRT {
  log.Info("Detected 0 clients, disabling real-time mode")
  atomic.StoreInt64(&l.realTimeEnabled, 0)
} else if !curEnabled && shouldEnableRT {
  log.Info("Detected active clients, enabling real-time mode")
  atomic.StoreInt64(&l.realTimeEnabled, 1)
}
```

Reply:
Once we see an enable we break out of the loop so it shouldn't care about the next disable. I think the tests are covering this case since it can be confusing. Your logic does look simpler and easier to understand so I'm happy to switch it something like that

Reply:
Ah, I see -- I missed the `break`!

Yeah, this took me a few minutes to grok, so might be worth simplifying